R/conditional-transform.R

```diff
@@ -0,0 +1,215 @@
+#' Conditional transformation
+#'
+#' Create a new variable that has values when specific conditions are met. 
+#' Conditions are specified using a series of formulas: the righthand side is 
+#' the condition that must be true (a `CrunchLogicalExpr`) and the lefthand 
+#' side is where to get the value if the condition on the righthand side is 
+#' true. This is commonly a Crunch variable.
+#'
+#' `conditionalTransform` is similar to `makeCaseVariable` however 
+#' `conditionalTransform` can use other Crunch variables as a source of a 
+#' variable, whereas, whereas `makeCaseVariable` can only use characters. This 
+#' additional power comes at a cost: `makeCaseVariable` can be executed 
+#' entirely on Crunch servers, so no data needs to be downloaded or uploaded 
+#' to/from the local R session. `conditionalTransform` on the other hand will 
+#' download the data necessary to construct the new variable.
+#'
+#' @param ... a list of cases to evaluate as well as other
+#' properties to pass about the case variable (i.e. alias, description)
+#' @param data a Crunch dataset object to use
+#' @param else_condition a default value to use if none of the conditions are 
+#' true (default: `NA`)
+#' @param type a character that is either "categorical", "text", "numeric" what
+#'  type of output should be returned? The source variables will be converted 
+#'  to this type if necessary
+#' @param categories if `type="categorical"`, these are all of the categories 
```

Comment:
What type of object is `categories`? In the example below, it appears to be character (category names, I presume), but that isn't the same thing as `Categories`. Fine to accept character in order to specify order of categories, but if this argument is going to exist, it should (also) accept full category definitions so that you can specify more category metadata (like missingness, numeric values, etc.)